### PR TITLE
Mi 67 code fix for lamps

### PR DESCRIPTION
**Pull request by RyanJField**
Fixes the need to store the codes.txt file locally. Allowing MetaInsight to run on the LAMPS.

Simply moves the code file to the download handler. The codes.txt is not used anywhere else.

### CODE REVIEW

**Review by JanionNevill — APPROVED** [review]
Looks fine

File: R/data/data_input_panel.R [hunk 1]
```diff
@@ -153,11 +153,42 @@ data_input_panel_server <- function(id, continuous_file = 'data/Non_opioids_long
...
+          )
+        )
+        invalid_data(df)
+        output$file_input_panel <- default_file_input
+        df <- defaultD()
+        data_uploaded(FALSE)
+      } else {
+        invalid_data(NULL)
+      }
+
```
Comment (JanionNevill) [inline R/data/data_input_panel.R:186]: This section was removed on dev. It was removed to correct a merge error on my part.

File: R/data/data_input_panel.R [hunk 2]
```diff
@@ -153,11 +153,42 @@ data_input_panel_server <- function(id, continuous_file = 'data/Non_opioids_long
...
+      } else {
+        invalid_data(NULL)
+      }
+      
       return(CleanData(df))
     })
     
     output$invalid_data <- DT::renderDataTable({
-      return(invalid_data())
+      invalid_data()
```
Comment (JanionNevill) [inline R/data/data_input_panel.R:191]: The return should be added back in

File: R/data/data_validation.R [hunk 3]
```diff
@@ -45,11 +45,6 @@ ValidateUploadedData <- function(data, outcome_type) {
     return(result)
   }
 
-  result <- .ValidateSingleArmStudies(data)
-  if (!result$valid) {
-    return(result)
-  }
-
```
Comment (JanionNevill) [inline R/data/data_validation.R:52]: Everything missing from this file should be reinstated.

File: tests/testthat/test-data_validation.R [hunk 4]
```diff
@@ -204,39 +204,3 @@ test_that("ValidateUploadedData() identifies invalid binary wide data, with inco
...
-})
-
-test_that("ValidateUploadedData() identifies invalid binary long data, with single-arm studies", {
-  data <- CleanData(read.csv("data/invalid_data/binary-single-arm-long.csv"))
-  
-  validation_result <- ValidateUploadedData(data, "Binary")
-  
-  expect_false(validation_result$valid)
-  expect_equal(validation_result$message, "Some studies have single arms: Constantine, Justinian")
-})
```
Comment (JanionNevill) [inline tests/testthat/test-data_validation.R:242]: Tests should be reinstated

File: tests/testthat/data/invalid_data/continuous-single-arm-wide.csv
Comment (JanionNevill) [inline tests/testthat/data/invalid_data/continuous-single-arm-wide.csv:1]: File should be reinstated

File: tests/testthat/data/invalid_data/continuous-single-arm-long.csv
Comment (JanionNevill) [inline tests/testthat/data/invalid_data/continuous-single-arm-long.csv:1]: File should be reinstated

File: tests/testthat/data/invalid_data/binary-single-arm-wide.csv
Comment (JanionNevill) [inline tests/testthat/data/invalid_data/binary-single-arm-wide.csv:1]: File should be reinstated

File: tests/testthat/data/invalid_data/binary-single-arm-long.csv
Comment (JanionNevill) [inline tests/testthat/data/invalid_data/binary-single-arm-long.csv:1]: File should be reinstated

**Review by JanionNevill — CHANGES_REQUESTED** [review]
Changes look good, I haven't run them to check though. I'll do that once the merge clashes have been addressed.

File: R/data/data_input_panel.R (hunk 1 above)
Comment (RyanJField) [reply R/data/data_input_panel.R]: Done :)

File: R/data/data_input_panel.R (hunk 2 above)
Comment (RyanJField) [reply R/data/data_input_panel.R]: Done :)

File: R/data/data_validation.R (hunk 3 above)
Comment (RyanJField) [reply R/data/data_validation.R]: Done :)

File: tests/testthat/test-data_validation.R (hunk 4 above)
Comment (RyanJField) [reply tests/testthat/test-data_validation.R:242]: Done :)

File: tests/testthat/data/invalid_data/continuous-single-arm-wide.csv
Comment (RyanJField) [reply tests/testthat/data/invalid_data/continuous-single-arm-wide.csv:1]: Done :)

File: tests/testthat/data/invalid_data/continuous-single-arm-long.csv
Comment (RyanJField) [reply tests/testthat/data/invalid_data/continuous-single-arm-long.csv:1]: Done :)

File: tests/testthat/data/invalid_data/binary-single-arm-wide.csv
Comment (RyanJField) [reply tests/testthat/data/invalid_data/binary-single-arm-wide.csv:1]: Done :)

File: tests/testthat/data/invalid_data/binary-single-arm-long.csv
Comment (RyanJField) [reply tests/testthat/data/invalid_data/binary-single-arm-long.csv:1]: Done :)

**Review by JanionNevill — APPROVED** [review]
Everything looks to be working 👍

**Pull request merged**